### PR TITLE
Move Java code from referer-parser/java-scala to it's own project

### PR DESCRIPTION
**Pull request by wiktorolko**
The code is based on https://github.com/snowplow-referer-parser/referer-parser/pull/156. All projects/ source files not related to Java were removed.

Additionally following minor changes has been introduced:
 * gradle as the build tool (version updated to 0.4.0)
 * make unit tests pass by subdomain parsing
 * minor warnings cleanup

### CODE REVIEW

**Comment by snowplowcla** [comment]
Thanks for your pull request. Is this your first contribution to a Snowplow open source project? Before we can look at your pull request, you'll need to sign a Contributor License Agreement (CLA).

:memo: Please visit https://github.com/snowplow/snowplow/wiki/CLA to learn more and sign.

Once you've signed, please reply here (e.g. I signed it!) and we'll verify. Thanks.

**Comment by wiktorolko** [comment]
I signed it!

**Comment by snowplowcla** [comment]
Confirmed! @wiktorolko  has signed the Individual Contributor License Agreement. Thanks so much

File: README.md [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+
+1. Fork it
+2. Create your feature branch (`git checkout -b my-new-feature`)
+3. Commit your changes (`git commit -am 'Add some feature'`)
+4. Push to the branch (`git push origin my-new-feature`)
+5. Create new Pull Request
+
+## Copyright and license
+
+Licensed under the [Apache License, Version 2.0][license] (the "License");
```
Comment (BenFradet) [inline README.md:76]: the license link is missing

**Review by BenFradet — COMMENTED** [review]
Thanks, just a minor comment in the readme and we're good to go :+1: .

Also once this is merged, feel free to open a pr removing it from the referer-parser repo!

**Comment by BenFradet** [comment]
As part of [hacktoberfest](https://hacktoberfest.digitalocean.com/), we have a raffle to win Snowplow t-shirts.
To enter the raffle you can fill out this form: https://docs.google.com/forms/d/e/1FAIpQLSeBbwQMWD9w9CEvjKt1KblDVpMy9mHOvQXS6YMRYH3geM8pCw/viewform

**Pull request merged**